code/modules/mob/living/carbon/human/species_types/golems.dm

```diff
@@ -551,18 +551,21 @@
 	icon_icon = 'icons/mob/actions/actions_spells.dmi'
 	var/cooldown = 150
 	var/last_teleport = 0
+	var/is_charging = FALSE //Set to true upon action activation to prevent spamming teleport callbacks while the first is still occurring.
```

Comment:
```suggestion
	///Set to true upon action activation to prevent spamming teleport callbacks while the first is still occurring.
	var/is_charging = FALSE 
```

Reply:
autodoc format good